opm/simulators/wells/GasLiftSingleWell_impl.hpp

```diff
@@ -136,24 +136,24 @@ computeWellRates_(Scalar bhp, bool bhp_is_limited, bool debug_output ) const
     return {-potentials[this->oil_pos_],
             -potentials[this->gas_pos_],
             -potentials[this->water_pos_],
+            bhp,
             bhp_is_limited
     };
 }
 
 template<typename TypeTag>
 std::optional<typename GasLiftSingleWell<TypeTag>::Scalar>
 GasLiftSingleWell<TypeTag>::
-computeBhpAtThpLimit_(Scalar alq, bool debug_output) const
+computeBhpAtThpLimit_(Scalar alq, Scalar bhp, bool debug_output) const
 {
     OPM_TIMEFUNCTION();
-    const auto& groupStateHelper = this->simulator_.problem().wellModel().groupStateHelper();
-    auto bhp_at_thp_limit = this->well_.computeBhpAtThpLimitProdWithAlq(
+    auto bhp_at_thp_limit = this->well_.computeBhpAtThpLimitProdWithAlqUsingIPR(
```

Comment:
Maybe add a comment why we now are using `computeBhpAtThpLimitProdWithAlqUsingIPR()` instead of `computeBhpAtThpLimitProdWithAlq()` ?